src/repo.rs

```diff
@@ -688,8 +698,26 @@ impl DocumentInfo {
     /// Count patches since last save,
     /// returns whether there were any.
     fn note_changes(&mut self) -> bool {
-        // TODO: count patches somehow.
-        true
+        // TODO, Can we do this without a read lock?
+        // I think that if the changes update last_heads and
+        // we store `last_heads_since_note` we can get a bool out of this.
+        let count = {
+            let doc = self.document.read();
+            let changes = doc.automerge.get_changes(&self.last_heads);
+            tracing::trace!(
+                "last: {:?}, current: {:?}",
+                self.last_heads,
+                doc.automerge.get_heads()
+            );
+            //self.last_heads = doc.automerge.get_heads();
```

Comment:
:thinking:

Reply:
I think this can be removed, as it is replaced by https://github.com/automerge/automerge-repo-rs/pull/58/files#diff-8a44ff48a251d03d58e6d3c73e8664553e2aa7b2a925641dd6c3d2f81c0d2e11R769